alien4cloud/alien4cloud.go

```diff
@@ -0,0 +1,2169 @@
+		if err != nil {
+			return errors.Wrapf(err, "Unable to get A4C application topology for app %s and env %s\n", a4cCtx.AppID, a4cCtx.EnvID)
+		}
+	}
+	err := c.editA4CTopology(a4cCtx, topoEditorExecute)
+	if err != nil {
+		return errors.Wrapf(err, "UpdateComponentProperty : Unable to edit the topology of application '%s' and environment '%s'\n", a4cCtx.AppID, a4cCtx.EnvID)
+	}
+
+	return nil
+}
+
+// UpdateComponentProperty Update the property value of a component of an application
+func (c *a4cClient) UpdateComponentProperty(a4cCtx *TopologyEditorContext, componentName string, propertyName string, propertyValue string) error {
+
+	if a4cCtx == nil {
+		return errors.New("Context object must be defined")
+	}
+
+	topoEditorExecute := TopologyEditorUpdateNodeProperty{
+		TopologyEditorExecuteRequest: TopologyEditorExecuteRequest{
+			NodeName:            componentName,
+			PreviousOperationID: a4cCtx.PreviousOperationID,
+			OperationType:       a4cUpdateNodePropertyValueOperationJavaClassName,
+		},
+		PropertyName:  propertyName,
+		PropertyValue: propertyValue,
+	}
+
+	if a4cCtx.TopologyID == "" {
+		var err error
+		a4cCtx.TopologyID, err = c.getA4CTopologyID(a4cCtx.AppID, a4cCtx.EnvID)
+		if err != nil {
+			return errors.Wrapf(err, "Unable to get A4C application topology for app %s and env %s\n", a4cCtx.AppID, a4cCtx.EnvID)
+		}
+	}
+	err := c.editA4CTopology(a4cCtx, topoEditorExecute)
+
+	if err != nil {
+		return errors.Wrapf(err, "UpdateComponentProperty : Unable to edit the topology of application '%s' and environment '%s'\n", a4cCtx.AppID, a4cCtx.EnvID)
+	}
+
+	return nil
+}
+
+// UpdateCapabilityProperty Update the property value of a capability related to a component of an application
+func (c *a4cClient) UpdateCapabilityProperty(a4cCtx *TopologyEditorContext, componentName string, propertyName string, propertyValue string, capabilityName string) error {
+
+	if a4cCtx == nil {
+		return errors.New("Context object must be defined")
+	}
+
+	topoEditorExecute := TopologyEditorUpdateCapabilityProperty{
+		TopologyEditorExecuteRequest: TopologyEditorExecuteRequest{
+			NodeName:            componentName,
+			PreviousOperationID: a4cCtx.PreviousOperationID,
+			OperationType:       a4cUpdateCapabilityPropertyValueOperationJavaClassName,
+		},
+		PropertyName:   propertyName,
+		PropertyValue:  propertyValue,
+		CapabilityName: capabilityName,
+	}
+
+	if a4cCtx.TopologyID == "" {
+		var err error
+		a4cCtx.TopologyID, err = c.getA4CTopologyID(a4cCtx.AppID, a4cCtx.EnvID)
+		if err != nil {
+			return errors.Wrapf(err, "Unable to get A4C application topology for app %s and env %s", a4cCtx.AppID, a4cCtx.EnvID)
+		}
+	}
+
+	err := c.editA4CTopology(a4cCtx, topoEditorExecute)
+
+	if err != nil {
+		return errors.Wrapf(err, "Unable to edit the topology of application '%s' and environment '%s'", a4cCtx.AppID, a4cCtx.EnvID)
+	}
+
+	return nil
+}
+
+// AddNodeInA4CTopology Add a new node in the A4C topology
+// not used any more
```

Comment:
Remove either the comment or the function

Reply:
Removed the comment